### PR TITLE
pytorch: 9.5 and 9.6

### PR DESCRIPTION
**Pull request by ypandya**
*Description of changes:*

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

File: chapter_recurrent-modern/encoder-decoder.md [hunk 1]
```diff
@@ -25,6 +25,20 @@ class Encoder(nn.Block):
         raise NotImplementedError
 ```
 
+```{.python .input  n=1}
```
Comment (AnirudhDagar) [inline chapter_recurrent-modern/encoder-decoder.md:28]: Looks like the tabs were not cleared out. Can you restart and clear the outputs of the notebook and then run `d2lbook activate all`.

This should look like -> ```{.python .input}

File: chapter_recurrent-modern/machine-translation-and-dataset.md [hunk 2]
```diff
@@ -118,13 +131,27 @@ def build_array(lines, vocab, num_steps, is_source):
...
+```{.python .input}
+#@tab pytorch
+#@save
+def build_array(lines, vocab, num_steps, is_source):
+    lines = [vocab[l] for l in lines]
+    if not is_source:
+        lines = [[vocab['<bos>']] + l + [vocab['<eos>']] for l in lines]
+    array = torch.tensor([truncate_pad(
+        l, num_steps, vocab['<pad>']) for l in lines])
+    valid_len = (array != vocab['<pad>']).sum(axis=1)
```
Comment (AnirudhDagar) [inline chapter_recurrent-modern/machine-translation-and-dataset.md:143]:
```suggestion
    valid_len = (array != vocab['<pad>']).sum(dim=1)
```
Let's use `dim` instead of `axis` just like the PyTorch documentation which uses `dim` almost everywhere.

File: d2l/torch.py [hunk 3]
```diff
@@ -670,6 +670,114 @@ def load_data_time_machine(batch_size, num_steps, use_random_iter=False,
...
+
+
+# Defined in file: ./chapter_recurrent-modern/machine-translation-and-dataset.md
+def build_array(lines, vocab, num_steps, is_source):
+    lines = [vocab[l] for l in lines]
+    if not is_source:
+        lines = [[vocab['<bos>']] + l + [vocab['<eos>']] for l in lines]
+    array = torch.tensor([truncate_pad(
+        l, num_steps, vocab['<pad>']) for l in lines])
+    valid_len = (array != vocab['<pad>']).sum(axis=1)
```
Comment (AnirudhDagar) [inline d2l/torch.py:723]: `dim`

File: chapter_recurrent-modern/encoder-decoder.md (hunk 1 above)
Comment (AnirudhDagar) [reply chapter_recurrent-modern/encoder-decoder.md:28]: @ypandya please address these as well.

Comment (ypandya) [reply chapter_recurrent-modern/encoder-decoder.md:28]: I did follow these instructions, there was no change here even after clearing and activating. I checked that there are many such lines in most of the existing notebooks for the mxnet portions. These do not get updated by clearing the output. This is not a part of the output but a part of the input of the cell. Due to this many code cells have some number on them when we open even without any output.

Comment (AnirudhDagar) [reply chapter_recurrent-modern/encoder-decoder.md:28]: Yeah, it is definitely only the cell number so that is cool. But I saw that these change to nil when we restart and clear outputs. Anyways will merge this as Mu has already approved the PR.

**Comment by AnirudhDagar** [comment]
Thanks for making the changes @ypandya :)
